src/views/virtualmachines/actions/components/CloneVMModal/utils/helpers.tsx

```diff
@@ -172,3 +177,70 @@ export const updateClonedDataVolumes = (
   vm.spec.template.spec.volumes = updatedVolumes;
   return vm;
 };
+
+const ControllerRevisionModel = {
+  kind: 'ControllerRevision',
+  apiGroup: 'apps',
+  apiVersion: 'v1',
+  abbr: 'CR',
+  label: 'ControllerRevision',
+  labelPlural: 'ControllerRevisions',
+  plural: 'controllerrevisions',
+};
+
+export const cloneControllerRevision = async (
+  revisionName: string,
+  destinationNamespace: string,
+  originNamespace: string,
+) => {
+  try {
+    return await k8sGet<IoK8sApiAppsV1ControllerRevision>({
+      model: ControllerRevisionModel,
+      name: revisionName,
+      ns: destinationNamespace,
+    });
+  } catch (error) {
+    if (error.code !== 404) {
+      throw error;
```

Comment:
You are throwing exceptions but not handling them on the other files when you use it...
I would have handled it here and reduced the possibility someone won't add a try-catch block or concating a .catch

Reply:
The error would be caught by TabModal and it would display the error with the nice UI.

Reply:
here I'm trying to fetch a ControllerVersion in another namespace using the referenceName.
I'll create it only if it's not there.

The fact that k8sGet throws a 404 error, its not actually an error. It just means that does not exist in that namespace.

Reply:
Probably in the next version, the backend will handle cloning in different namespaces and we'll remove all the logic. But for now we need that

Reply:
im sorry, but it isn't making sense to me. Please clarify in slack, thank you :)